src/main.rs

```diff
@@ -214,15 +226,13 @@ async fn main() {
         });
     }
 
-    // Exit on Ctrl-C (SIGINT) and SIGTERM.
-    let mut term_signal = unix_signal(SignalKind::terminate()).unwrap();
-
-    tokio::select! {
-        _ = signal::ctrl_c() => (),
-        _ = term_signal.recv() => (),
-    };
+    // initiate graceful shutdown sequence on sig int
+    let mut stream = unix_signal(SignalKind::interrupt()).unwrap();
 
-    info!("Shutting down...");
+    stream.recv().await;
+    client::SHUTTING_DOWN.store(true, Ordering::Relaxed);
+    warn!("Got SIGINT, waiting for client connection drain now");
+    wg.wait();
```

Comment:
I'm wondering if just saving `tokio::task::spawn` handles in a `Vec` in `main.rs` and then waiting for them all to exit `task.await` is best?